dataplane/forwarding/fwdport/ports/kernel.go

```diff
@@ -158,11 +160,22 @@ func (kernelBuilder) Build(portDesc *fwdpb.PortDesc, ctx *fwdcontext.Context) (f
 	if !ok {
 		return nil, fmt.Errorf("invalid port type in proto")
 	}
+	l, err := netlink.LinkByName(kp.Kernel.GetDeviceName())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get interface: %v", err)
+	}
+	// Since the TAP port and external ports have different MACs and thus DST_MAC may be not correct,
+	// read all packets for processing.
+	if l.Attrs().Promisc == 0 {
```

Comment:
I still don't quite understand, I think I'm missing the overall picture. I'll have to look at the other parts of the code again, but if it's working this LGTM to submit.

Reply:
i don't know that is strictly required, but it can help troubleshoot issues if packets have an incorrect destination.